Sources/UIAlertControllerView.swift

```diff
@@ -43,48 +57,100 @@ class UIAlertControllerView: UIView {
             return button
         }
 
+        clipsToBounds = true
         backgroundColor = .white
         layer.cornerRadius = 3
 
-        header.sizeToFit()
-        header.frame.height += verticalPadding
-        header.layer.contentsGravity = .top
         addSubview(header)
+        text.map { addSubview($0) }
+        buttons.forEach { addSubview($0) }
+
+        subviewWidth = {
+            let minWidth: CGFloat = 300
+            let maxReadableWidth: CGFloat = 540
+            let maxWidth: CGFloat
+            if UIScreen.main.isPortrait {
+                maxWidth = min(UIScreen.main.bounds.width - 4 * horizontalPadding, maxReadableWidth)
+            } else {
+                maxWidth = min(max(minWidth, UIScreen.main.bounds.width / 1.5), maxReadableWidth)
+            }
 
-        buttons.forEach{ addSubview($0) }
+            subviews.forEach { $0.sizeToFit() }
+            guard let largestSubviewWidth = subviews.map({ $0.frame.width }).max() else {
+                assertionFailure("no subviews exist to determine largestSubviewWidth")
+                return minWidth
+            }
+            return min(maxWidth, max(largestSubviewWidth, minWidth))
+        }()
+        
+        subviews.forEach { $0.clipsToBounds = true }
     }
 
     override func layoutSubviews() {
-        switch style {
-        case .actionSheet: layoutAsActionSheet()
-        default: assertionFailure(
-            "The UIAlertControllerStyle style, \(style), is not implemented yet!")
+        header.frame.origin = CGPoint(x: horizontalPadding, y: verticalPadding)
+        header.bounds.width = subviewWidth
+        header.sizeToFit()
+        header.textAlignment = .left // sizeToFit() seems to change textAlignment
```

Comment:
can you elaborate on this comment? This sounds like a bug in our code. Or does it happen on iOS as well?

Reply:
I checked this again. It seems it was an unnecessary comment, since 
```
header.textAlignment = .left
header.sizeToFit()
```
seems to work as expected. changing it.